devtools/klap-decrypt.py

```diff
@@ -0,0 +1,272 @@
+"""
+This code allow for the decryption of KlapV2 data from a pcap file.
+
+It will output the decrypted data to a file.
+This was designed and tested with a Tapo light strip setup using a cloud account.
+"""
+
+import codecs
+import json
+import re
+
+import pyshark
+from cryptography.hazmat.primitives import padding
+
+from kasa.credentials import Credentials
+from kasa.deviceconfig import (
+    DeviceConfig,
+    DeviceConnectionParameters,
+    DeviceEncryptionType,
+    DeviceFamily,
+)
+from kasa.klaptransport import KlapEncryptionSession, KlapTransportV2
+
+# These are the only variables that need to be changed in order to run this code.
+# ********** CHANGE THESE **********
+username = "hello@world.com"  # cloud account username (likely an email)
+password = "hunter2"  # cloud account password  # noqa: S105
+# the ip of the smart device as it appears in the pcap file
+device_ip = "192.168.1.100"
+pcap_file_path = "/path/to/my.pcap"  # the path to the pcap file
+output_json_name = (
+    "output.json"  # the name of the output file, relative to the current directory
+)
+# **********************************
+
+capture = pyshark.FileCapture(pcap_file_path, display_filter="http")
+
+# In an effort to keep this code tied into the original code
+# (so that this can hopefully leverage any future codebase updates inheriently),
+# some weird initialization is done here
+myCreds = Credentials(username, password)
+
+fakeConnection = DeviceConnectionParameters(
+    DeviceFamily.SmartTapoBulb, DeviceEncryptionType.Klap
+)
+fakeDevice = DeviceConfig(
+    device_ip, connection_type=fakeConnection, credentials=myCreds
+)
+
+
+# In case any modifications need to be made in the future,
+# this class is created to allow for easy modification
+class MyKlapTransport(KlapTransportV2):
+    """A custom KlapTransportV2 class that allows for easy modification."""
+
+    pass
```

Comment:
```suggestion
```
I would skip this for now. We can adapt when and if necessary :-)

Reply:
Yeah, I don't remember why I thought that was a good idea lol. I think it was an artifact of my original revision, at I for some reason just went with it.

Reply:
Fixed manually in 83e512fb65df123f42be3e39557d12cb9ccc338c